cmd/activator/request_log_test.go

```diff
@@ -91,13 +94,37 @@ func TestUpdateRequestLogFromConfigMap(t *testing.T) {
 		body:     "test",
 		template: "",
 		want:     "",
+	}, {
+		name:             "explicitly enable request logging",
+		url:              "http://example.com/testpage",
+		body:             "test",
+		template:         "{{.Request.URL}}\n",
+		want:             "http://example.com/testpage\n",
+		enableRequestLog: ptr.Bool(true),
+	}, {
+		name:             "explicitly disable request logging",
+		url:              "http://example.com/testpage",
+		body:             "test",
+		template:         "disable_request_log",
+		want:             "",
+		enableRequestLog: ptr.Bool(false),
+	}, {
+		name:             "explicitly enable request logging with empty template",
+		url:              "http://example.com/testpage",
+		body:             "test",
+		template:         "",
+		want:             "",
+		enableRequestLog: ptr.Bool(true),
```

Comment:
could also just explicitly put the `data` map in the test table, i.e. add `data map[string]string` as a field in the table, and then you can explicitly show the keys there/not there in each test. e.g. a row might look like

~~~~
{
		name: "explicitly disable request logging",
		url:  "http://example.com/testpage",
		body: "test",
		data: map[string]string{
			"logging.template":           "disable_request_log",
			"logging.enable-request-log": "false",
		},
		want: "",
 }
~~~~

avoids the ptr stuff and the if block/dereference below, and is maybe a bit clearer? I don't feel super strongly though if you prefer this way

Reply:
let me check :)